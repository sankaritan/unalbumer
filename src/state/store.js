import { createStore, applyMiddleware, combineReducers, compose } from "redux";
import { routerReducer, routerMiddleware } from "react-router-redux";
import createSagaMiddleware from "redux-saga";
import createHistory from "history/createBrowserHistory";
import thunk from "redux-thunk";
import { rootReducer, rootSaga } from "./ducks";

export const history = createHistory();
history.push("/");

// initial state of the store, that will be used by the app
export const initialState = {
  user: {
    username: null,
    loggedIn: false,
    oauthToken: null
  },
  photos: {
    albums: [],
    photosInAlbums: [],
    allPhotos: [],
    newAlbum: null
  }
};

const sagaMiddleware = createSagaMiddleware();
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  combineReducers({
    ...rootReducer,
    router: routerReducer
  }),
  initialState,
  composeEnhancers(
    applyMiddleware(thunk, routerMiddleware(history), sagaMiddleware)
  )
);

sagaMiddleware.run(rootSaga);

export default store;
