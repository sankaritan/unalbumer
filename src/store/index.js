import { createStore, applyMiddleware, combineReducers, compose } from "redux";
import { routerReducer, routerMiddleware } from "react-router-redux";
import createHistory from "history/createBrowserHistory";
import thunk from "redux-thunk";
import { rootReducer } from "../reducers/rootReducer";

export const history = createHistory();
history.push("/");

// initial state of the store, that will be used by the app
export const initialState = {
  user: {
    username: null,
    loggedIn: false,
    oauthToken: null,
  },
  photos: {
    albums: [],
    photosInAlbums: [],
    allPhotos: [],
  }
};

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  combineReducers({
    ...rootReducer,
    router: routerReducer
  }),
  initialState,
  composeEnhancers(applyMiddleware(thunk, routerMiddleware(history)))
);

export default store;
