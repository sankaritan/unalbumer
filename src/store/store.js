import { createStore, applyMiddleware, combineReducers, compose } from "redux";
import { routerReducer, routerMiddleware } from "react-router-redux";
import createHistory from "history/createBrowserHistory";
import { rootReducer } from "../reducers/rootReducer";
import thunk from "redux-thunk";

export const history = createHistory();
history.push("/");

// initial state of the store, that will be used by the app
export const initialState = {
  counter: {
    count: 0,
    isIncrementing: false,
    isDecrementing: false
  },
  homeTitle: "Default Home Title",
  somethingElse: true
};

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

// create store object
const store = createStore(
  // combine all custom reducers + router reducer into one
  combineReducers({
    ...rootReducer,
    router: routerReducer
  }),
  initialState,
  // define middleware
  // > thunk - to allow actions that return function
  // > history - to allow routing via pushing into history array)
  composeEnhancers(applyMiddleware(thunk, routerMiddleware(history)))
);

export default store;
