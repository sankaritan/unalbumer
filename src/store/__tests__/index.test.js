import store from "..";

describe("Store", () => {
  it("PageHeader renders correctly", () => {
    expect(store).toMatchSnapshot();
  });
});
