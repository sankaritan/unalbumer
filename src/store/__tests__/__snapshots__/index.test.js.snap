// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store PageHeader renders correctly 1`] = `
Object {
  "dispatch": [Function],
  "getState": [Function],
  "replaceReducer": [Function],
  "subscribe": [Function],
  Symbol(observable): [Function],
}
`;
